tools/ideabuck/src/com/facebook/buck/intellij/ideabuck/run/BuckRunConfigurationExtension.java

```diff
@@ -0,0 +1,207 @@
+package com.facebook.buck.intellij.ideabuck.run;
+
+import com.facebook.buck.intellij.ideabuck.build.BuckBuildManager;
+import com.facebook.buck.intellij.ideabuck.build.BuckCommand;
+import com.facebook.buck.intellij.ideabuck.build.BuckQueryCommandHandler;
+import com.facebook.buck.intellij.ideabuck.build.ResultCallbackBuckHandler;
+import com.facebook.buck.intellij.ideabuck.config.BuckModule;
+import com.facebook.buck.intellij.ideabuck.file.BuckFileUtil;
+import com.google.common.base.Function;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.intellij.execution.BeforeRunTask;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.RunConfigurationExtension;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.application.ApplicationConfiguration;
+import com.intellij.execution.configurations.JavaParameters;
+import com.intellij.execution.configurations.RunConfigurationBase;
+import com.intellij.execution.configurations.RunnerSettings;
+import com.intellij.execution.impl.RunManagerImpl;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiClass;
+import com.intellij.util.PathsList;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.jetbrains.annotations.NotNull;
+
+public class BuckRunConfigurationExtension extends RunConfigurationExtension {
+
+  private List<String> targetResults;
+  static private File tempFile;
+
+  @Override
+  public <T extends RunConfigurationBase> void updateJavaParameters(T configuration,
+      JavaParameters javaParameters, RunnerSettings runnerSettings) throws ExecutionException {
+    Project currentProject = configuration.getProject();
+    VirtualFile containingFile;
+    VirtualFile buckFile;
+    if (configuration instanceof ApplicationConfiguration) {
+      containingFile = ((ApplicationConfiguration) configuration).getMainClass()
+          .getContainingFile().getVirtualFile();
+      buckFile = BuckFileUtil.getBuckFile(containingFile);
+    } else {
+      return;
+    }
+    List<String> additionalClasspath = new ArrayList<>();
+    BuckBuildManager buildManager = BuckBuildManager.getInstance(currentProject);
+    BuckQueryCommandHandler buckQueryCommandHandler =
+        getBuckQueryCommandHandler(currentProject, buckFile);
+    buckQueryCommandHandler.command().addParameter("owner(" + containingFile.getPath() + ")");
```

Comment:
Should this query also restrict the `buck.type` to one of the `java_*` targets, so that it doesn't apply to arbitrary targets like `python_library` or even `genrule`?